components/feature/autofill/src/main/java/mozilla/components/feature/autofill/ui/AbstractAutofillSearchActivity.kt

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.autofill.ui
+
+import android.app.assist.AssistStructure
+import android.content.Intent
+import android.os.Build
+import android.os.Bundle
+import android.view.autofill.AutofillManager
+import android.widget.EditText
+import androidx.annotation.RequiresApi
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.FragmentActivity
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.storage.Login
+import mozilla.components.feature.autofill.AutofillConfiguration
+import mozilla.components.feature.autofill.R
+import mozilla.components.feature.autofill.response.dataset.LoginDatasetBuilder
+import mozilla.components.feature.autofill.structure.ParsedStructure
+import mozilla.components.feature.autofill.structure.parseStructure
+import mozilla.components.feature.autofill.structure.toRawStructure
+import mozilla.components.feature.autofill.ui.search.LoginsAdapter
+import mozilla.components.support.ktx.android.view.showKeyboard
+
+/**
+ * Activity responsible for letting the user manually search and pick credentials for auto-filling a
+ * third-party app.
+ */
+@RequiresApi(Build.VERSION_CODES.O)
+abstract class AbstractAutofillSearchActivity : FragmentActivity() {
+    abstract val configuration: AutofillConfiguration
+
+    private lateinit var parsedStructure: ParsedStructure
+    private lateinit var loginsDeferred: Deferred<List<Login>>
+    private val scope = CoroutineScope(Dispatchers.IO)
+    private val adapter = LoginsAdapter(::onLoginSelected)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val structure: AssistStructure? =
+            intent.getParcelableExtra(AutofillManager.EXTRA_ASSIST_STRUCTURE)
+        if (structure == null) {
+            finish()
+            return
+        }
+
+        val parsedStructure = parseStructure(this, structure.toRawStructure())
+        if (parsedStructure == null) {
+            finish()
+            return
+        }
+
+        this.parsedStructure = parsedStructure
+        this.loginsDeferred = loadAsync()
+
+        setContentView(R.layout.mozac_feature_autofill_search)
```

Comment:
nit: should everything below this be in `onCreateView`?

Reply:
You mean `onViewCreated()`? That only exists in fragments afaik.

Reply:
Oops, yeah I forgot about that. 😄